### PR TITLE
ucp: Change error when directory is specified but not recursive

### LINKED ISSUE #10537: omitting `-r` when copying a directory with `ucp` gives unhelpful error
### Describe the bug

as in the title :yum: 

### How to reproduce

1. create a directory with a file in it
```nushell
mkdir foo
"foo" | save foo/foo.foo
```
2. try to copy the directory without `-r`
```nushell
> ucp foo/ bar
-nu: omitting directory '/whatever/foo'
```

### Expected behavior

i expected something more like the builtin `cp` command :yum:
```nushell
> core cp foo/ bar
Error:   × Directories must be copied using "--recursive"
   ╭─[entry #10:1:1]
 1 │ core cp foo/ bar
   ·         ──┬─
   ·           ╰── resolves to a directory (not copied)
   ╰────
```

### Screenshots

_No response_

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.85.1                                   |
| branch             | main                                     |
| commit_hash        | 8f4ea69c22dc9d2d1b86499330f095a76c8c569d |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.70.0 (90c541806 2023-05-31)      |
| rust_channel       | 1.70.0-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.70.0 (ec8a8a0ca 2023-04-25)      |
| build_time         | 2023-09-27 18:45:31 +02:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  | gstat, nu_plugin_explore                 |

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dmatos2012**
# Description
Closes #10537. Basically error message was unhelpful, and this temporary measure adds back the nice previous nushell error message.   Ideally, we would like to add a more permanent solution mentioned in the issue [comments](https://github.com/nushell/nushell/issues/10537#issuecomment-1743686122), but since we want to have `ucp` as `cp` on new release, this is hackier but way simpler so this fix should do it. 

Only downside is that now behavior differs from `uutils` in the sense that:
```
uutils:
> cp a foo/ bar
ls bar
# foo/a

nushell:
>ucp a foo/ bar
# directory error (not copied) ....
```
So, since its non fatal error, uutils copies a, but nushell errors out with nothing copied. If we go to option 3 mentioned above, then we can decide what we want to do, and perhaps continue on a non fatal error. 

# User-Facing Changes

# Tests + Formatting
Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- [X] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [X] `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used` to check that you're using the standard code style
- [X] `cargo test --workspace` to check that all tests pass (on Windows make sure to [enable developer mode](https://learn.microsoft.com/en-us/windows/apps/get-started/developer-mode-features-and-debugging))
- [X] `cargo run -- -c "use std testing; testing run-tests --path crates/nu-std"` to run the tests for the standard library

> **Note**
> from `nushell` you can also use the `toolkit` as follows
> ```bash
> use toolkit.nu  # or use an `env_change` hook to activate it automatically
> toolkit check pr
> ```

# After Submitting

### CODE REVIEW

File: crates/nu-command/src/filesystem/ucp.rs [hunk 1]
```diff
@@ -166,6 +166,17 @@ impl Command for UCp {
                         if f.is_empty() {
                             return Err(ShellError::FileNotFound(p.span));
                         }
+                        let any_source_is_dir = f.iter().any(|f| matches!(f, f if f.is_dir()));
+                        if any_source_is_dir && !recursive {
+                            return Err(ShellError::GenericError(
+                                "Directories must be copied using \"--recursive\"".into(),
+                                "resolves to a directory (not copied)".into(),
+                                Some(p.span),
+                                None,
```
Comment (amtoine) [inline crates/nu-command/src/filesystem/ucp.rs:175]: may i propose
```suggestion
                                "could_not_copy_directory".into(),
                                "resolves to a directory (not copied)".into(),
                                Some(p.span),
                                Some("Directories must be copied using \"--recursive\"".into()),
```

to get an error like
```
Error:   × could_not_copy_directory
   ╭─[entry #2:1:1]
 1 │ ucp foo/ bar
   ·     ──┬─
   ·       ╰── resolves to a directory (not copied)
   ╰────
  help: Directories must be copied using "--recursive"
```
? :innocent:

Comment (dmatos2012) [reply crates/nu-command/src/filesystem/ucp.rs]: Yes, I like your error better. Changed :)

Comment (amtoine) [reply crates/nu-command/src/filesystem/ucp.rs]: amazing thanks :pray: :star_struck:

**Comment by amtoine** [comment]
thanks @dmatos2012 for the quick and great fix :pray:

**Pull request merged**